File: java/src/binary_search/BinarySearchFindLastTest.java
package binary_search;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class BinarySearchFindLastTest {
	@Test
	public void testRandomCases() {
		for (var test = 0; test < 1_000; test++) {
			var td = new randomTestData();

			var expected = td.expectedLast;
			var actual = BinarySearchFindLast.findLastInt(td.array, td.val);
			if (actual != expected) {
				Assert.fail(
					String.format("findLastInt(%s, %s):\n  Actual: %s\nExpected: %s",
						Arrays.toString(td.array),
						td.val,
						actual,
						expected
					)
				);
			}
		}
	}
}

File: go/binary_search/binary_search_find_last_test.go
package binary_search

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFindLast_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.expectedLast
		actual := findLastInt(td.array, td.val)
		if actual != expected {
			assert.FailNowf(t, "",
				"findLastInt(%v, %v):\n  Actual: %v\nExpected: %v",
				td.array, td.val, actual, expected,
			)
		}
	}
}
